secure_message/common/utilities.py

```diff
@@ -70,68 +70,75 @@ def process_paginated_list(paginated_list, host_url, user, message_args, endpoin
-    """looks up the details for the from users"""
-    logger.info("Getting user from details", is_internal=message['from_internal'], user_uuid=message['msg_from'])
-    if message['from_internal']:
-        from_details = internal_user_service.get_user_details(message['msg_from'])
-    else:
-        from_details = party.get_user_details(message['msg_from'])
-    return from_details
+def get_from_details(messages):
+    """Adds a @msg_from key every message in a list of messages.
+    Every msg_to uuid is resolved to include details of the user.
 
+    If the call for the internal user id fails, an exception will be thrown.
+    If the external user id cannot be found in the list that we got from the party service.  There
+    won't be a @msg_from value returned in the payload.  The API documentation notes that these elements
+    aren't guaranteed to be provided so we're not breaking the contract by doing this.
+    """
+    external_user_details = party.get_users_details(get_external_user_uuid_list(messages))
+    for message in messages:
+        if message["from_internal"]:
+            message.update({"@msg_from": internal_user_service.get_user_details(message["msg_from"])})
+        else:
+            for detail in external_user_details:
+                if detail['id'] == message['msg_from']:
+                    message.update({'@msg_from': detail})
+                    break
+    return messages
 
-def _get_to_details(message):
-    """looks up the details of all the to users"""
-    user_details = []
-    if message['from_internal']:
-        to_user_details_finder = party.get_user_details
-    else:
-        to_user_details_finder = internal_user_service.get_user_details
-
-    for uuid in message['msg_to']:
 
-        detail = to_user_details_finder(uuid)
+def get_external_user_uuid_list(messages):
+    """Compiles a list of all unique the external user (respondant) uuids from a list of messages"""
+    external_user_uuids = []
 
-        if detail:
-            user_details.append(detail)
+    external_msgs = [message for message in messages if message['from_internal'] is False]
+    for uuid in external_msgs:
+        if uuid["msg_from"] not in external_user_uuids:
+            external_user_uuids.append(uuid["msg_from"])
 
-    return user_details
+    internal_messages = [message for message in messages if message['from_internal'] is True]
+    for uuid in internal_messages:
+        if uuid["msg_to"][0] not in external_user_uuids:
+            external_user_uuids.append(uuid["msg_to"][0])
+    return external_user_uuids
 
 
 def add_business_details(messages):
-    """Adds business details"""
-
-    ru_list = []
+    """Adds a @ru_id key every message in a list of messages."""
+    ru_ids = []
 
     for message in messages:
-        if message['ru_id'] not in ru_list:
-            ru_list.append(message['ru_id'])
+        if message['ru_id'] not in ru_ids:
+            ru_ids.append(message['ru_id'])
```

Comment:
Same comment as previously about using a set to avoid the need for the `if`.

Reply:
Done